### PR TITLE
Added unit tests for actions and stores

### PR DESCRIPTION
**Pull request by jedireza**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: tests/unit/actions/showDoc.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            key: '/docs/quick-start.md'
+        };
+
+        context.executeAction(showDoc, params, function (err) {
+            if (err) {
+                return done(err);
+            }
+
+            var docs = context.getStore(DocStore).getAll();
+            expect(docs).to.be.an('object');
```
Comment (redonkulus) [inline tests/unit/actions/showDoc.js]: Can we add another assertion on the data returned?

File: tests/unit/actions/showDoc.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
```
Comment (redonkulus) [inline tests/unit/actions/showDoc.js]: How do we know this is from the cache?

File: tests/unit/stores/ApplicationStore.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+/* global describe, it, beforeEach */
+'use strict';
+var expect = require('chai').expect;
+var ApplicationStore = require('../../../stores/ApplicationStore');
+var routesConfig = require('../../../configs/routes');
+
+describe('doc store', function () {
```
Comment (redonkulus) [inline tests/unit/stores/ApplicationStore.js]: application store

File: tests/unit/stores/ApplicationStore.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    it('should handle navigate', function (done) {
+        storeInstance.handleNavigate(homeRoute);
+
+        expect(storeInstance.currentPageId).to.equal(homeRoute.params.key);
+        done();
+    });
+
+    it('should return early during handle navigate, when navigating to the same page', function (done) {
+        storeInstance.handleNavigate(homeRoute);
+        storeInstance.handleNavigate(homeRoute);
```
Comment (redonkulus) [inline tests/unit/stores/ApplicationStore.js]: duplicate

Comment (jedireza) [inline tests/unit/stores/ApplicationStore.js]: Nope. We navigate to the same place twice on purpose.

Comment (redonkulus) [inline tests/unit/stores/ApplicationStore.js]: can you add a comment just to be clear?

**Comment by redonkulus** [comment]
Add copyright comments to new files.

File: tests/unit/actions/showDoc.js (hunk 1 above)
Comment (jedireza) [inline tests/unit/actions/showDoc.js]: Now we create the context and mock service before each test. I've made it more clear that we're executing the action again. Also confirmed with code coverage report.

**Comment by redonkulus** [comment]
:shipit:

**Pull request merged**